src/main/java/frc/robot/subsystems/drivetrain/Chassis.kt

```diff
@@ -482,27 +478,124 @@ object Chassis :
                 hasPoseTarget = false
             }
 
+    private val pathConstraintZones =
+        listOf(
+            // Between approach point and final target, limit velocity to 0.5 meters per
+            // second.
+            // The other gains are arbitrarily high, since it likely doesn't matter what
+            // those values are
+            ConstraintsZone(1.0, 2.0, PathConstraints(0.5, 5.0, 10.0, 100.0, 12.0))
+        )
+    private val pathEventMarkers = listOf(EventMarker("atApproachPoint", 0.8))
+    private val pathConstraints =
+        PathConstraints(
+            4.metersPerSecond,
+            4.metersPerSecondPerSecond,
+            1.67.rotationsPerSecond,
+            3.rotationsPerSecondPerSecond,
+            12.volts,
+        )
+
+    /**
+     * Drives to a pose using pathplanner. The robot will always approach the target pose in the x
+     * direction.
+     *
+     * @param approachBackward If true, the robot will approach the target pose in the -x direction.
+     * @param pose The target pose to drive to.
+     */
+    fun pathplanToPose(
+        approachBackward: Boolean = true,
+        approachDistance: Distance = 0.5.meters,
+        pose: () -> Pose2d,
+    ): Command = defer {
+        val currentPose = state.Pose
+        val targetPose = pose()
+        // Approach the target from 0.5 meter back
+        val approachPoint =
+            targetPose.transformBy(
+                Transform2d(
+                    // If approaching the target 'backward', approach from .5 meters in front of the
+                    // robot, else, .5 meters behind the robot
+                    if (approachBackward) approachDistance else -approachDistance,
+                    0.meters,
+                    // If we're approaching the target 'backward', then the direction of travel
+                    // should be 180º, i.e the back of the robot should be leading the path
+                    if (approachBackward) Rotation2d.k180deg else Rotation2d.kZero,
+                )
+            )
+        val waypoints =
+            PathPlannerPath.waypointsFromPoses(
+                Pose2d(
+                    currentPose.translation,
+                    // Start the path in the direction of the approach point
+                    (approachPoint.translation - currentPose.translation).angle,
```

Comment:
Important to use approach point here, for the case when the robot starts between approach point and target. We want the robot to back up to re-align